File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F19.P2.cpp
#include <atomic>
#include <cstdint>
#include <cstdio>
#include <thread>

std::int64_t shared_total() {
// BEGIN PROBE F19.P2
std::atomic_int64_t counter{0};
auto bump = [&]{ for (int i = 0; i < 1000; ++i) ++counter; };
std::thread t1(bump);
std::thread t2(bump);
t1.join();
t2.join();
return counter;
// END PROBE F19.P2
}

int main() {
    std::printf("%lld\n", shared_total());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F19.P2.ts
// BEGIN PROBE F19.P2
async function run() {
  const counter = new BigInt64Array(new SharedArrayBuffer(8));
  const bump = async () => {
    for (let i = 0; i < 1000; i++) Atomics.add(counter, 0, 1n);
  };
  await Promise.all([bump(), bump()]);
  return counter[0];
}
// END PROBE F19.P2

run().then(v => { console.log(String(v)); });
